### DISCUSSION TITLE
include only root dependencies in result

### DISCUSSION THREAD

**Discussion by DmitriyLewen** [aquasecurity/trivy#4870] [Ideas]
### Description

Trivy includes all indirect dependencies for `pom.xml` files.
But some users asked about report only direct dependencies (See #4840)
 
I suggest discuss this options:
- how to correctly exclude indirect dependencies
- how many people need this option
- etc...

### Target

Container Image

### Scanner

Vulnerability

**Comment by knqyf263** [comment]
I guess the dependency graph for POM addresses this confusion. They get to know how transitive dependencies are introduced.

**Comment by ba1ma0** [comment]
We need it ! Strong support this option !!!! Thank you ~👍🏻👍🏻👍🏻👍🏻

**Comment by zangcc** [comment]
I hope Trivy can add an option to report only direct dependencies. Thank you~❤️ @DmitriyLewen

**Comment by kkmantianxing** [comment]
Hello, I think the original Trivy function can be retained, but I think it is necessary to add this option to scan direct dependencies. From a security point of view, the previous scan results are obviously more comprehensive, but the attack and exploitation chain for dependencies is very complicated, let alone indirect dependencies. Based on the above, I would prefer to add this option. Thanks.

**Comment by RexHarrr** [comment]
Hi, I hope this option can be added, I use trivy a lot at work, thanks.

**Comment by ivdr7873843** [comment]
Hello, I think it is good to add this option, I hope to add it as soon as possible, after all, I have encountered the same problem when scanning.

**Comment by glebignatieff** [comment]
+1

**Comment by DmitriyLewen** [comment]
Hi all, we added [--pkg-relationships](https://trivy.dev/latest/docs/scanner/vulnerability/#filtering-by-package-relationships) flag.
It may help you.
